File: backend/src/main/java/com/hear2speak/dtos/AppointmentSearchRequest.java
package com.hear2speak.dtos;

import java.time.LocalDateTime;

import com.hear2speak.entities.AppointmentStatus;

public class AppointmentSearchRequest {

    // sort parameters

    public String sortField;

    public Boolean ascending;

    // search parameters

    public String globalText;

    public LocalDateTime startDateFrom;

    public LocalDateTime startDateTo;
    
    public AppointmentStatus appointmentStatus;

}

File: frontend/src/app/interfaces/appointment/appointment-search-request.ts
import { AppointmentStatus } from "./appointment-status";

export interface AppointmentSearchRequest {

    // sort parameters

    sortField: string | null;
    ascending: boolean | null;

    // search parameters

    globalText: string | null;
    startDateFrom: string | null;
    startDateTo: string | null;
    appointmentStatus: AppointmentStatus | null;

    // pagination

    page: number | null;
    size: number | null;

}